### PR TITLE
Movable text in board notes

### PR DESCRIPTION
**Pull request by krisx303**
New toolbar button for adding new text component
When new component is added then it is in editable state (what means that textinput appers at the place of text - there is place to improvement but it works)
When single click on text component (also images) then component is in moving state
When double click on text/image component then it is in editable state
Refactored Board component by extracting many helper functions and changed logic in updating list of elements (previous idea was really bad in case of react performance...)

There is still lack of edition options of text like changing the font size / color

### CODE REVIEW

**Comment by Adam3004** [comment]
is it set like we discussed offline? I mean one click to move and two clicks to edit?

File: src/pages/notes/board/MoveableText.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const MovableText = ({
+      font, element
+}: MovableTextProps) => {
+
+    return (
+        <>
+            {(element.isEditingMode || element.isMoving) && <Path
+                path={Skia.Path.MakeFromSVGString(svgRectangleBorder(
+                    element.position.x - 5, element.position.y, element.content.length * 10, element.height))!}
```
Comment (Adam3004) [inline src/pages/notes/board/MoveableText.tsx:19]: why -5 and *10?

File: src/pages/notes/board/TextInputComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    top: element.position.y - 15,
+                    width: 200, // Adjust width as needed
+                    height: 30, // Adjust height as needed
+                    opacity: 1, // Keep this TextInput somewhat transparent
+                    backgroundColor: "white",
+                }}
+                value={newText}
+                onChangeText={setNewText}
+                onBlur={() => {element.content = newText!;element.isEditingMode = false;
+                    props.setElements(props.genericMovableElements.map(e => e.id === element.id ? element : e));}} // Stop editing when losing focus
```
Comment (Adam3004) [inline src/pages/notes/board/TextInputComponent.tsx:30]: great comment here

**Review by Adam3004 — APPROVED** [review]

File: src/pages/notes/board/MoveableText.tsx (hunk 1 above)
Comment (krisx303) [reply src/pages/notes/board/MoveableText.tsx:19]: Currently there is no option to change the font size of the text and `10` is static value for one single letter. So we are multipling the length of the word times `10`. The `-5` is for drawing the rectangle border in the some radius from the text. (btw I do not know why we don't have to add `1` to content.length value but it works as expected xD)

**Pull request merged**